### PR TITLE
Installers

### PR DESCRIPTION
**Pull request by Sw1fty**

### CODE REVIEW

**Comment by mcksn** [comment]
@vswarnkar there are extra changes in your commit. are those intended?

File: Installers/Mac/Setup/Contents/Info.plist [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        <key>CFBundleSignature</key>
+        <string>????</string>
+        <key>CFBundleVersion</key>
+        <string>1.101</string>
+        <key>NSHumanReadableCopyright</key>
+        <string>Copyright (C) 2022</string>
+        <key>NSHighResolutionCapable</key>
+        <string>true</string>
+        <key>NSMicrophoneUsageDescription</key>
+        <string>The application minima is requesting access to the microphone.</string>
```
Comment (mcksn) [inline Installers/Mac/Setup/Contents/Info.plist:30]: scary. is this needed?

Comment (vswarnkar) [reply Installers/Mac/Setup/Contents/Info.plist:30]: I'll remove this, it isn't required. Have added during a testing phase. thanks!

**Comment by vswarnkar** [comment]
> @vswarnkar there are extra changes in your commit. are those intended?

They were pushed for reference to create another installer for different version

File: Installers/Mac/Setup/Contents/MacOS/minima [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    osascript \
+	-e "set question to display dialog \"$ERROR_MSG\" with title \"$ERROR_TITLE\" buttons {\"Cancel\", \"Download\"} default button 2" \
+	-e "if button returned of question is equal to \"Download\" then open location \"$DOWNLOAD_URL\""
+	echo "$ERROR_TITLE"
+	echo "$ERROR_MSG"
+	exit 1
+fi
+
+if [ ! -d "$HOME/.minima" ]; then
+	mkdir $MINIMA_FOLDER
```
Comment (mcksn) [inline Installers/Mac/Setup/Contents/MacOS/minima:33]: which user is executing this script? if it's the user, then we can allow minima.jar to create this automatically right

File: Installers/Mac/Setup/Contents/MacOS/minima [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+DOWNLOAD_URL="https://www.oracle.com/uk/java/technologies/javase/jdk11-archive-downloads.html"
+
+# Is Java installed?
+if type -p java; then
+    _java="java"
+elif [[ -n "$JAVA_HOME" ]] && [[ -x "$JAVA_HOME/bin/java" ]]; then
+    _java="$JAVA_HOME/bin/java"
+else
+    osascript \
+	-e "set question to display dialog \"$ERROR_MSG\" with title \"$ERROR_TITLE\" buttons {\"Cancel\", \"Download\"} default button 2" \
```
Comment (mcksn) [inline Installers/Mac/Setup/Contents/MacOS/minima:25]: is this a dialog box that shows up?

File: Installers/Mac/Setup/Contents/MacOS/minima [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+	chmod u+x .minima
+fi
+
+if [ ! -f "$MINIMA_FOLDER/minima.sh" ]; then
+	cd $MINIMA_FOLDER
+
+	cp $DIR/../Resources/minima.sh $MINIMA_FOLDER
+	chmod +x $MINIMA_FOLDER/minima.sh
+	
+	echo 'alias minima='$MINIMA_FOLDER'/minima.sh' >>~/.bash_profile
```
Comment (mcksn) [inline Installers/Mac/Setup/Contents/MacOS/minima:43]: is it possible to put this on the binary path instead?
not common practice to setup aliases in people's profiles I dont think
would make uninstall awkward too

File: Installers/Mac/Setup/Contents/MacOS/minima (hunk 2 above)
Comment (vswarnkar) [reply Installers/Mac/Setup/Contents/MacOS/minima:33]: Yes - I've updated script as part of latest PR https://github.com/minima-global/Minima/pull/570 (I wanted to create this to copy the minima script that runs the curl command internally)

**Comment by mcksn** [comment]
app icon is low res? anything we can do about this

File: Installers/Mac/Setup/Contents/MacOS/minima (hunk 2 above)
Comment (vswarnkar) [reply Installers/Mac/Setup/Contents/MacOS/minima:33]: This was added to copy the minima script that has curl command into `.minima` directory. I've removed this now in my latest PR: https://github.com/minima-global/Minima/pull/570/

File: Installers/Mac/Setup/Contents/MacOS/minima (hunk 3 above)
Comment (vswarnkar) [reply Installers/Mac/Setup/Contents/MacOS/minima:25]: That's correct, if JAVA_HOME couldn't find a popup would have been displayed. Removed in: https://github.com/minima-global/Minima/pull/570/

File: Installers/Mac/Setup/Contents/MacOS/minima (hunk 4 above)
Comment (vswarnkar) [reply Installers/Mac/Setup/Contents/MacOS/minima:43]: I've updated installer to open in a terminal window so this isn't required

File: Installers/Mac/Setup/Contents/Resources/minima.sh [hunk 5]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+
+old="$IFS"
```
[the remark this answers is not in the document]
Comment (vswarnkar) [reply Installers/Mac/Setup/Contents/Resources/minima.sh:3]: > @vswarnkar app icon is low res? anything we can do about this

I can ask Pete to provide an icon as current icon was converted by myself from png to icn format

**Comment by vswarnkar** [comment]
> @vswarnkar any reason for the ~3 mins of the app bouncing on application tray? maybe java download..but then a dialog box would have been presented right?

Won't be there in new installer: https://github.com/minima-global/Minima/pull/570/

**Comment by Sw1fty** [comment]
FYI - I'm need to sort out just merging in the installer files. There is a bunch of stuff in here that shouldn't be merged

**Pull request closed without merging**